Extensions/TileMap/pixi-tilemap-helper.js

```diff
@@ -169,6 +168,151 @@
     return tileMapData;
   };
 
+  /**
+   * Parse a Tiled map Ldtk file,
+   * exported from Ldtk (https://ldtk.io/json/)
+   * into a generic tile map data (`GenericPixiTileMapData`).
+   *
+   * @param {Object} tiledData A JS object representing a map exported from Tiled.
+   * @param {?PIXI.BaseTexture} atlasTexture
+   * @param {(textureName: string, relativeToPath: string) => PIXI.BaseTexture} getTexture A getter to load a texture. Used if atlasTexture is not specified.
+   * @returns {?GenericPixiTileMapData}
+   */
+  const parseLDtkData = (
+    tiledData,
+    atlasTexture,
+    getTexture,
+    levelIndex,
+    tilemapResourceName
+  ) => {
+    const tileSetAtlases = {};
+    tiledData.defs.tilesets.forEach((tileset) => {
+      const texture = tileset.relPath
+        ? getTexture(tileset.relPath, tilemapResourceName)
+        : null;
+      tileSetAtlases[tileset.uid] = { texture, ...tileset };
+    });
+    const selectedLevel = tiledData.levels[levelIndex > -1 ? levelIndex : 0];
+
+    const layers = [];
+    const textureCache = {};
+    selectedLevel.layerInstances.reverse().forEach((ldtkLayer, layerIndex) => {
+      const layerAtlasTextureRelPath = ldtkLayer['__tilesetRelPath'];
+      const gridSize = ldtkLayer['__gridSize'];
+      const type = ldtkLayer['__type'];
+      const tilesetUid = ldtkLayer['__tilesetDefUid'];
+      const autoLayerTiles = ldtkLayer.autoLayerTiles;
+      const gridTiles = ldtkLayer.gridTiles;
+      const ldtkTiles = [...autoLayerTiles, ...gridTiles];
+      const entities = ldtkLayer.entityInstances || [];
+      const layer = {
+        type: type,
+        autoLayerTiles,
+        ldtkTiles,
+        gridTiles,
+        entityInstances: entities,
+        visible: ldtkLayer.visible,
+        opacity: ldtkLayer['__opacity'],
+      };
+      textureCache[layerIndex] = {};
+      const tileSet = tileSetAtlases[tilesetUid];
+
+      ldtkTiles.forEach((generatedTile) => {
+        if (generatedTile.t in textureCache[layerIndex]) return;
+
+        try {
+          const [x, y] = generatedTile.src;
+          const rect = new PIXI.Rectangle(x, y, gridSize, gridSize);
+
+          // @ts-ignore - atlasTexture is never null here.
+          const texture = new PIXI.Texture(tileSet.texture, rect);
+
+          textureCache[layerIndex][generatedTile.t] = texture;
+        } catch (error) {
+          console.error(
+            'An error occurred while creating a PIXI.Texture to be used in a TileMap:',
+            error
+          );
+          textureCache[layerIndex] = null;
+        }
+      });
+
+      layers.push(layer);
+    });
+
+    const selectedLevelBg = selectedLevel.bgRelPath;
+    textureCache.levelBg = {};
+    if (selectedLevelBg) {
+      if (selectedLevelBg in textureCache.levelBg) return;
```

Comment:
Can you explain what this early return is doing?

Reply:
it prevents it from caching it, if its already cached